platform/migration/platform-migratable/src/main/java/org/codice/ddf/migration/impl/PlatformMigratable.java

```diff
@@ -0,0 +1,216 @@
+/**
+ * Copyright (c) Codice Foundation
+ * <p/>
+ * This is free software: you can redistribute it and/or modify it under the terms of the GNU Lesser
+ * General Public License as published by the Free Software Foundation, either version 3 of the
+ * License, or any later version.
+ * <p/>
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.ddf.migration.impl;
+
+import static org.apache.commons.lang.Validate.notNull;
+
+import java.io.IOException;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.Collection;
+
+import javax.validation.constraints.NotNull;
+
+import org.apache.commons.io.FileUtils;
+import org.apache.commons.lang.StringUtils;
+import org.codice.ddf.migration.Migratable;
+import org.codice.ddf.migration.MigrationException;
+import org.codice.ddf.migration.MigrationMetadata;
+import org.codice.ddf.migration.MigrationWarning;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * This class handles the export process for all Platform configurations and system files.
+ *
+ */
+public class PlatformMigratable implements Migratable {
+
+    private static final Logger LOGGER = LoggerFactory
+            .getLogger(PlatformMigratable.class);
+
+    private static final String KEYSTORE_SYSTEM_PROP = "javax.net.ssl.keyStore";
+
+    private static final String TRUSTSTORE_SYSTEM_PROP = "javax.net.ssl.trustStore";
+
+    private static final String WS_SECURITY_DIR = "etc/ws-security";
+
+    private static final String SYSTEM_PROPERTIES = "etc/system.properties";
+
+    private static final String USERS_PROPERTIES = "etc/users.properties";
+    
+    private static final String ABSOLUTE_PATH_WARNING =
```

Comment:
Since some of those messages may be used by other `Migratable`implementations, do you think we should move them to the `Migratable` interface? Or create `MigrationWarning` sub-classes that take the property name and path in the constructor?